### ISSUE TITLE
INDEX should not be able to construct pseudoinstructions

### ISSUE THREAD

**Issue by thewonderidiot** [virtualagc/virtualagc#1066]
While working on improving peripheral instruction support for my AGC Monitor clone today, I discovered an interesting quirk of pseudoinstruction testing. For whatever reason, it's impossible to construct RELINT, INHINT, or EXTEND using INDEX. If you attempt attempt to do so, the computer will actually execute TC 3, TC 4, or TC 6, respectively, rather than their usual pseudoinstruction operation.

This happens because of how pseudoinstruction testing is timed. The test is performed whenever T7PHS4/ and TSUDO/ are both low:
![image](https://user-images.githubusercontent.com/94056/56463261-7cf8ef80-6385-11e9-9de1-07295f01129f.png)

T7PHS4/ is low during phase 4 of timepulse 7 (unless, on 2003993-021 or later AGCs, FUTEXT is set indicating an extracode). TSUDO/ is constructed as follows:
![image](https://user-images.githubusercontent.com/94056/56463309-51c2d000-6386-11e9-9146-d1dee32f3282.png)

Note that IC3 is set during TC0, TCF0, and STD2, and IC16 is set during BZF0 and BZMF0 if they do not take their branch. This covers all instructions that can load new instructions _except_ INDEX -- meaning that the test for pseudoinstructions is not performed when INDEX is the active instruction. Thus, INDEX can be used to construct TC 3, TC 4, and TC 6, but the lack of a test for them means that they will be executed as regular instructions.

**Comment by rburkey2005** [comment]
Incidentally, anybody who hasn't seen episode 8 of the AGC restoration 
videos <https://www.youtube.com/watch?v=ThOXgcOOA1M> on youtube should 
do so.  In it, Mike demonstrates using his AGC monitor, and it's pretty 
amazing to watch.  Of course, in the video, you only see the monitor 
working with Mike's FPGA clone of the AGC, but given the accuracy of 
Mike's reconstruction of the AGC I presume it's no different (with 
proper connectorization and electrical buffering) on the real AGC.

On a different note, check out this new article on the AGC in American 
Scientist <https://www.americanscientist.org/article/moonshot-computing> 
(no, not Scientific American).  I provided a little input to it, so 
Virtual AGC gets a nice mention in it. Even some of the old AGC 
developers took note of it, so you know it has the official seal of 
approval on it. :-)

-- Ron

**Comment by smithery1** [comment]
I know this is pretty old, but just to note that this interesting fact is also mentioned in the AGC4 Basic Training Manual [document](http://www.ibiblio.org/apollo/NARA-SW/E-2052.pdf) in section 1.4.3, "Op-Code Selection Logic". It doesn't shed any light on the "why", but it discusses INDEX (and more obviously EXTEND) inhibiting any special code that follows, and contains a flow diagram illustrating the selection steps. It even has a small code example that shows the hacky developer a way they can, in fact, transfer to 4.

**Comment by Milek7** [comment]
How does that work when interrupt is taken after pseudoinstruction was already fetched? Wouldn't instruction code be stored in 15-bit BRUPT, which would make it impossible to disambiguate pseudoinstructions after returning from interrupt?

**Comment by thewonderidiot** [comment]
> I have found partial explanation in ND-1021042: "If c(G) = 000003 (RELINT), 000004 (INHINT) or 000006 (EXTEND) control pulse RAD causes the generation of pulses RZ and ST2 and subinstruction STD2 is executed next. If G contains any other quantity control pulse RAD causes the generation of control pulse RG and next instruction is executed". That is, interrupt cannot be taken there because pseudoinstructions work by effectively extending instruction that fetched them with extra STD2 cycle.

Yep, that's exactly correct -- interrupts cannot occur if the next instruction to be executed is a pseudoinstruction.

> That still leaves the question of RESUME: E-2052 claims it also cannot be constructed with INDEX, but it's executed more conventionally by NDX0->RSM3 and I cannot see why it would behave this way.

The easy answer here is that E-2052 is wrong. I just ran the following code in our AGC gate-by-gate simulator:
```
		CAF	RSM2
		INDEX	A
		INDEX	A
		TC	+2
RSM2		OCT	17
```
The second INDEX executed as NDX0->RSM3:
![image](https://user-images.githubusercontent.com/94056/223301727-41bcf127-94d7-4ba9-8e80-be210f0daa66.png)
